test/e2e/v3/e2e_suite.go

```diff
@@ -67,7 +67,7 @@ var _ = Describe("kubebuilder", func() {
 			err := kbc.Init(
 				"--project-version", "3-alpha",
 				"--domain", kbc.Domain,
-				"--dep=false")
+				"--fetch-deps=false")
```

Comment:
I may miss something, but why do we switch to use the new flag? What's the motiviation?

Reply:
In some point, we received a suggestion to remove the deprecated flags. However, I revert this suggestion. It is not part of the scope of this PR. Here we are using removing the V1 code. So, good catcher. It is using dep again.

Reply:
Is `—dep` not deprecated? It says so in code comments. Was this depreciation unrelated to the v2 change?

Reply:
it is deprecated but still valid for v2. 
We can remove the flag for V3 since is deprecated and I think we should do that. 
However, it shows not part of the scope of this one and then, I think that would be better we have a specific or for that.